src/kOS/Suffixed/WaypointValue.cs

```diff
@@ -8,47 +9,80 @@ namespace kOS.Suffixed
-            AddSuffix("NEARSURFACE", new NoArgsSuffix<bool>(() => wayPoint.isOnSurface, "True if waypoint is a point near or on the body rather than high in orbit."));
-            AddSuffix("GROUNDED", new NoArgsSuffix<bool>(() => wayPoint.landLocked, "True if waypoint is actually glued to the ground."));
+            AddSuffix("NEARSURFACE", new NoArgsSuffix<bool>(() => WrappedWaypoint.isOnSurface, "True if waypoint is a point near or on the body rather than high in orbit."));
+            AddSuffix("GROUNDED", new NoArgsSuffix<bool>(() => WrappedWaypoint.landLocked, "True if waypoint is actually glued to the ground."));
+            AddSuffix("INDEX", new NoArgsSuffix<int>(() => WrappedWaypoint.index, "Number of this waypoint if this is a grouped waypoint (i.e. alpha/beta/gamma.."));
+            AddSuffix("CLUSTERED", new NoArgsSuffix<bool>(() => WrappedWaypoint.isClustered, "True if this is a member of a cluster of waypoints (i.e. alpha/beta/gamma.."));
+            // greekMap is static, so whichever waypoint instance's constructor happens to
+            // get called first will make it, and from then on other waypoints don't need to
+            // keep re-initializing it:
+            if (greekMap == null)
+                InitializeGreekMap();
+        }
+        
+        private static void InitializeGreekMap()
+        {
+            greekMap = new Dictionary<string, int>();
+            for (int i = 0 ; i < 20 ; ++i)
+                greekMap.Add(FinePrint.Utilities.StringUtilities.IntegerToGreek(i).ToLower(), i);
         }
         
         public CelestialBody GetBody()
         {
-            return CachedBody ?? (CachedBody = VesselUtils.GetBodyByName(WayPoint.celestialName));
+            return CachedBody ?? (CachedBody = VesselUtils.GetBodyByName(WrappedWaypoint.celestialName));
         }
 
         public GeoCoordinates BuildGeoCoordinates()
         {
-            return new GeoCoordinates(GetBody(), Shared, WayPoint.latitude, WayPoint.longitude);
+            return new GeoCoordinates(GetBody(), Shared, WrappedWaypoint.latitude, WrappedWaypoint.longitude);
+        }
+        
+        public Vector GetPosition()
+        {
+            return new Vector(GetBody().GetWorldSurfacePosition(WrappedWaypoint.latitude, WrappedWaypoint.longitude, BuildSeaLevelAltitude()));
         }
         
         public double BuildSeaLevelAltitude()
         {
             GeoCoordinates gCoord = BuildGeoCoordinates();
-            return gCoord.GetTerrainAltitude() + WayPoint.altitude;
+            return gCoord.GetTerrainAltitude() + WrappedWaypoint.altitude;
         }
         
         public override string ToString()
         {
-            return String.Format("Waypoint \"{0}\"", WayPoint.name);
+            return String.Format("Waypoint \"{0}\"", CookedName() );
+        }
+        
+        public string CookedName()
+        {
```

Comment:
The purpose of ```CookedName()``` (which is also accessed via the suffix ```:NAME``` is to present the user with the facade that the waypoint's name really is the name they see on the screen.